packages/core/src/renderer/getStyleSheetForBucket.ts

```diff
@@ -33,11 +34,19 @@ export const styleBucketOrdering: StyleBucketName[] = [
  */
 export function getStyleSheetForBucket(
   bucketName: StyleBucketName,
-  target: Document,
+  target: Document | undefined,
   renderer: GriffelRenderer,
   elementAttributes: Record<string, string> = {},
-): CSSStyleSheet {
+): IsomorphicStyleSheet {
   if (!renderer.styleElements[bucketName]) {
+    const tag: HTMLStyleElement | undefined = target && target.createElement('style');
+    const stylesheet = createIsomorphicStyleSheet(tag, bucketName, elementAttributes);
+    renderer.styleElements[bucketName] = stylesheet;
+
+    if (!target || !tag) {
```

Comment:
```suggestion
    if (!tag) {
```

We are checking for `target` on line 42, right?